source/material_model/visco_plastic.cc

```diff
@@ -654,6 +657,8 @@ namespace aspect
                   else
                     out.reaction_terms[i][this->introspection().compositional_index_for_name("viscous_strain")] = e_ii;
                 }
+              if (track_noninitial_plastic_strain == true && plastic_yielding == true)
```

Comment:
@anne-glerum - That is what I originally did, but thought it might be useful to track the plastic strain even in cases where strain weakening is not used? Thoughts?

Reply:
After a bit more thought my preference would be to keep `track_noninitial_plastic_strain` separate from strain-weakening. I just tried using the option with the continental extension cookbook (no strain softening) and the `noninitial_plastic_strain` field nicely illustrates the evolution of the rift basin, including where the ductile layers are located. 

Another scenario to use this variable is with the finite strain tensor, which works with total strain. One may want to do strain-softening with the finite strain tensor invariant, but still track the time-dependent location of faults.

Are you comfortable moving forward with the current structure?